src/resources/auth/middleware/Roles.ts

```diff
@@ -0,0 +1,219 @@
+export type Role = {
+	readonly role: string;
+	readonly allowedRoutes: ReadonlyArray<{
+		readonly action: string;
+		readonly restrictions?: readonly string[];
+	}>;
+};
+
+export const admin = "admin";
+export const editor = "editor";
+export const author = "author";
+export const member = "member";
+export const unassigned = "unassigned";
```

Comment:
Hier könnten wir auch den existierenden `Role`-Type weiterverwenden, um Typensicherheit zu haben. Und dann könnten wir uns auch die ganzen separaten String-Definitionen sparen:

```suggestion
import { Role as RoleType } from "../../../generated/models/Role.generated";

export type Role = {
	readonly role: RoleType;
	readonly allowedRoutes: ReadonlyArray<{
		readonly action: string;
		readonly restrictions?: readonly string[];
	}>;
};
```

(Weiter unten muss dann `role: "admin"` statt `role :admin` verwendet werden usw.)

Reply:
Hier habe ich auf alle Fälle Role importiert und verwende es. Den definierten Typ Role habe ich umbenannt und nicht mehr exportiert. Die exportierten Konstanten habe ich gelassen, da ich diese extern verwende.